srml/contracts/src/wasm/prepare.rs

```diff
@@ -93,6 +93,50 @@ impl<'a> ContractModule<'a> {
 		Ok(())
 	}
 
+	/// Ensures that no floating point types are in use.
+	fn ensure_no_floating_types(&self) -> Result<(), &'static str> {
+		if let Some(global_section) = self.module.global_section() {
+			for global in global_section.entries() {
+				match global.global_type().content_type() {
+					ValueType::F32 | ValueType::F64 =>
+						return Err("use of floating point type in globals is forbidden"),
+					_ => {}
+				}
```

Comment:
Nit: It looks to me that `if let` would serve the purpose better. Also I am ok with using a macro/function using a less informative error message.